users/user.go

```diff
@@ -0,0 +1,177 @@
+package users
+
+import (
+	"fmt"
+	"log"
+	"net/url"
+	"strings"
+	"time"
+
+	"github.com/LF-Engineering/dev-analytics-libraries/elastic"
+
+	"github.com/LF-Engineering/dev-analytics-libraries/auth0"
+	"github.com/LF-Engineering/dev-analytics-libraries/http"
+
+	json "github.com/json-iterator/go"
+)
+
+// Auth0ClientProvider ...
+type Auth0ClientProvider interface {
+	GetToken() (string, error)
+}
+
+// HTTPClientProvider ...
+type HTTPClientProvider interface {
+	Request(string, string, map[string]string, []byte, map[string]string) (int, []byte, error)
+}
+
+// ESClientProvider ...
+type ESClientProvider interface {
+	CreateDocument(index, documentID string, body []byte) ([]byte, error)
+	Search(index string, query map[string]interface{}) ([]byte, error)
+	CreateIndex(index string, body []byte) ([]byte, error)
+	Get(index string, query map[string]interface{}, result interface{}) error
+}
+
+// SlackProvider ...
+type SlackProvider interface {
+	SendText(text string) error
+}
+
+// Usr struct
+type Usr struct {
+	UserBaseURL      string
+	ESCacheURL       string
+	ESCacheUsername  string
+	ESCachePassword  string
+	AuthGrantType    string
+	AuthClientID     string
+	AuthClientSecret string
+	AuthAudience     string
+	AuthURL          string
+	Environment      string
+	httpClient       HTTPClientProvider
+	auth0Client      Auth0ClientProvider
+	esClient         ESClientProvider
+	slackProvider    SlackProvider
+}
+
+// ListUsers ...
+func (u *Usr) ListUsers(email string, pageSize string, offset string) (*ListUsersResponse, error) {
```

Comment:
Since we already in users package no need for users suffix, so it's better to rename it to List instead of ListUsers

Reply:
OK, let me update

Reply:
Done @khalifapro